tests/pipelines/stable_diffusion_sag/test_stable_diffusion_sag.py

```diff
@@ -32,13 +32,15 @@
 from diffusers.utils.testing_utils import enable_full_determinism, nightly, require_torch_gpu, torch_device
 
 from ..pipeline_params import TEXT_TO_IMAGE_BATCH_PARAMS, TEXT_TO_IMAGE_IMAGE_PARAMS, TEXT_TO_IMAGE_PARAMS
-from ..test_pipelines_common import PipelineLatentTesterMixin, PipelineTesterMixin
+from ..test_pipelines_common import IPAdapterTesterMixin, PipelineLatentTesterMixin, PipelineTesterMixin
 
 
 enable_full_determinism()
 
 
-class StableDiffusionSAGPipelineFastTests(PipelineLatentTesterMixin, PipelineTesterMixin, unittest.TestCase):
+class StableDiffusionSAGPipelineFastTests(
```

Comment:
This pipeline will also fail tests because of old IP Adapter logic with `encode_image`. I think SAG is used a bit in community (atleast I make use of it sometimes), so it makes sense to support IP Adapters with latest implementation and tests here. WDYT? @yiyixuxu

Reply:
yes I agree :)